src/ophys_etl/modules/video/utils.py

```diff
@@ -624,19 +695,54 @@ def _video_array_from_h5(
     if reticle:
         for ii in range(d_reticle, video_shape[1], d_reticle):
             old_vals = np.copy(video_as_uint[:, ii:ii+2, :, :])
-            new_vals = np.zeros(old_vals.shape, dtype=np.uint8)
-            new_vals[:, :, :, 0] = 255
-            new_vals = (new_vals//2) + (old_vals//2)
-            new_vals = new_vals.astype(np.uint8)
+            new_vals = np.zeros(old_vals.shape, dtype=video_dtype)
+            new_vals[:, :, :, 0] = max_cast_value
+            new_vals = (new_vals//4) + (3*old_vals//4)
+            new_vals = new_vals.astype(video_dtype)
             video_as_uint[:, ii:ii+2, :, :] = new_vals
         for ii in range(d_reticle, video_shape[2], d_reticle):
             old_vals = np.copy(video_as_uint[:, :, ii:ii+2, :])
-            new_vals = np.zeros(old_vals.shape, dtype=np.uint8)
-            new_vals[:, :, :, 0] = 255
-            new_vals = (new_vals//2) + (old_vals//2)
-            new_vals = new_vals.astype(np.uint8)
+            new_vals = np.zeros(old_vals.shape, dtype=video_dtype)
+            new_vals[:, :, :, 0] = max_cast_value
+            new_vals = (new_vals//4) + (3*old_vals//4)
+            new_vals = new_vals.astype(video_dtype)
             video_as_uint[:, :, ii:ii+2, :] = new_vals
 
         logger.info('added reticles')
 
     return video_as_uint
+
+
+def apply_mean_filter_to_video(
+        video: np.ndarray,
+        kernel_size: int) -> np.ndarray:
+    """
+    Use skimage.measure.block_reduce to downsample a video
+    spatially by taking the mean of (kernel_size, kernel_size)
+    blocks
+
+    Parameters
+    ----------
+    video: np.ndarray
+        (ntime, nrows, ncols)
+
+    kernel_size: int
+
+    Returns
+    -------
+    downsampled_video: np.ndarray
+        (ntime, ceil(nrows/kernel_size), ceil(n_cols/kernel_size))
+
+    Notes
+    -----
+    If spatial dimensions are not exactly divisible by kernel_size,
+    the input array will be padded by zeros before the mean is taken
+    """
+
+    downsampled_video = skimage_measure.block_reduce(
+                            video,
```

Comment:
It's a bummer we can't combine our median filter with mean filter somehow. They are both doing the same thing except they use different functions and do/don't do downsampling

Reply:
Probably the responsible thing to do would be to consolidate these in a `video_utils` module. Do you mind if we leave that work fro #436 (if you agree, I'll add this as an explicit bullet to that ticket)?